src/core/js/models/itemsModel.js

```diff
@@ -0,0 +1,124 @@
+define([
+    'core/js/adapt',
+    'core/js/models/componentModel'
+], function(Adapt, ComponentModel) {
+
+    var ItemsModel = ComponentModel.extend({
+
+        reset: function(type, force) {
+            this.resetItems();
+            ComponentModel.prototype.reset.call(this, type, force);
+        },
+
+        resetItems: function() {
+            _.each(this.get('_items'), function(item) {
+                item._isVisited = false;
+                item._isActive = false;
+            });
+        },
+        
+        getItemCount: function() {
+            return this.get('_items').length;
+        },
+
+        getItemAtIndex: function(index) {
+            return this.get('_items')[index];
+        },
+
+        getVisitedItems: function() {
+            return _.filter(this.get('_items'), function(item) {
+                return item._isVisited;
+            });
+        },
+
+        setItemAtIndexAsVisited: function(index) {
+            var item = this.get('_items')[index];
+            item._isVisited = true;
+        },
+
+        getCompletionStatus: function() {
```

Comment:
Dunno about this. I can see the function name ``getCompletionStatus`` transcending the ``ItemsModel`` container as it doesn't really have to be related to the items abstract - except in this case. Then again, the behaviour the function contains is more aptly described by ``areAllItemsCompleted`` / ``getAllItemsCompleted``... so perhaps it is a good idea?